module/zfs/zfs_ioctl.c

```diff
@@ -3016,6 +3155,56 @@ zfs_ioc_set_fsacl(zfs_cmd_t *zc)
 	return (error);
 }
 
+#if defined(CONFIG_USER_NS) && defined(HAVE_KUIDGID_T)
+/*
+ * On success this consumes the nvlist in pnvp and replaces the pointer with
+ * one where ids have been mapped into the current user namespace.
+ * In case of error, pnvpn and its pointee stay unmodified.
+ */
+static int
+deleg_map_user_ns(nvlist_t **pnvp)
```

Comment:
This should probably be renamed `dsl_deleg_map_user_ns()` and moved it in to `dsl_deleg.c` along with the `fsacl_map_*` functions.  The code added to `zfs_ioc_set_fsacl` should probably also be there there as well in it's own function.

Let's also make sure that these functions exist and `return 0` when `CONFIG_USER_NS && HAVE_KUIDGID_T` aren't defined.  This should help keep this a little more readable and portable.